master/buildbot/process/builder.py

```diff
@@ -319,7 +320,7 @@ def run_cleanups():
 
         # set up locks
         build.setLocks(self.config.locks)
-        cleanups.append(lambda: slavebuilder.slave.releaseLocks())
+        cleanups.append(lambda: slavebuilder.slave.releaseLocks() if slavebuilder.slave is not None else None)
```

Comment:
There isn't an issue upstream for this. I suspect that this may be happening because commit 3d93216 sets slave to None when insubstantiation occurs and other places are not expecting it to be None. I don't remember seeing an issue like this until I introduced that patch. But, that's just a guess.